### PR TITLE
[mmp] A few framework support bug fixes

### PR DESCRIPTION
**Pull request by chamons**
- https://github.com/xamarin/xamarin-macios/issues/3973
- https://github.com/xamarin/xamarin-macios/issues/4066

### CODE REVIEW

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -1070,7 +1070,7 @@ static void HandleFramework (StringBuilder args, string framework, bool weak)
 			else
 				BuildTarget.Frameworks.Add (name);
 
-			if (!framework.EndsWith (".framework", StringComparison.Ordinal))
+			if (!framework.EndsWith (".framework", StringComparison.OrdinalIgnoreCase))
```
Comment (spouliot) [inline tools/mmp/driver.cs]: That seems incomplete
```
git grep "EndsWith (\".framework\""
```
returns many other cases (some shared my mtouch) where we check for path ending with `.framework`

Comment (chamons) [reply tools/mmp/driver.cs]: So the other one directly in mmp does:

```
if (v.EndsWith (".framework", true, CultureInfo.InvariantCulture))
```

with true being ignoreCase.

You are correct that mtouch doesn't have this fix, I could look at doing it in this PR, but I wouldn't want to write a test given the existing MSBuild testing situation.

Comment (spouliot) [reply tools/mmp/driver.cs]: I did not look if the code itself was used but the `common/*` files are also used by `mmp`

Comment (chamons) [reply tools/mmp/driver.cs]: I don't believe the common bits are needed to get the macOS use case working (given my test).

I don't want to change the ios\common bits without a test.

**Comment by chamons** [comment]
FYI - I'm considering this blocked until I can write tests for the iOS side that I'd be fixing, which I can't do without checking in additional project files, which I'd rather not do (or changing the default project to always use .Framework).

**Pull request closed without merging**